File: android/app/src/main/kotlin/in/nann/smashhockey/screens/RefusedScreen.kt
package `in`.nann.smashhockey.screens

import `in`.nann.smashhockey.core.generated.CopyKey
import `in`.nann.smashhockey.game.Game
import `in`.nann.smashhockey.game.L
import `in`.nann.smashhockey.generated.Presentation
import `in`.nann.smashhockey.ui.BlockButton
import `in`.nann.smashhockey.ui.Entrance
import `in`.nann.smashhockey.ui.Label3D
import `in`.nann.smashhockey.ui.Panel
import `in`.nann.smashhockey.ui.Paragraph
import `in`.nann.smashhockey.ui.WaveText

/**
 * A save that cannot be read (spec §15) — the twin of iOS's RefusedScreen.swift: the game says so
 * on a screen of its own and never opens as if the player were new. The file stays on the phone,
 * untouched, beside the new one; the one way on is to start over.
 */
class RefusedScreen(game: Game, why: String) :
    Screen(Game.pose(Presentation.Screens.Refused.eye, Presentation.Screens.Refused.target), game) {
    init {
        part(WaveText(kit, L(CopyKey.REFUSED_TITLE), 0.15f, C.CORAL, bob = 0.4f, id = "refused_title_header"), at(0f, top - 0.5f))
        val card = part(Panel(kit, 1.66f, 1.2f, 0.14f, C.CREAM, Entrance.Tumble), at(0f, top - 1.45f))
        child(Paragraph(kit, L(CopyKey.REFUSED_BODY), 0.07f, C.INK, 1.44f, id = "refused_body"), at(0f, 0.08f), card.content).show(0.0)
        // The reason, small, for whoever is asked to look at the phone.
        child(Label3D(kit, why, 0.035f, C.DISABLED_INK, maxWidth = 1.44f), at(0f, -0.45f), card.content).show(0.0)
        val go = part(BlockButton(kit, L(CopyKey.REFUSED_START_OVER), "refused_startover_button", BlockButton.Style.DANGER, 1.3f, 0.38f, 0.13f) {
            game.startOver()
        }, at(0f, bottom + 0.4f))
        go.bobs = true
    }
}

File: ios/Sources/Screens/RefusedScreen.swift
import RealityKit
import SmashCore

/// A save that cannot be read (spec §15): the game says so on a screen of its own and never opens
/// as if the player were new. The file stays on the phone, untouched, beside the new one; the one
/// way on is to start over. The twin of Android's RefusedScreen.kt.
@MainActor
final class RefusedScreen: Screen {
    init(game: Game, why: String) {
        super.init(pose: CameraPose(Presentation.Screens.Refused.eye, Presentation.Screens.Refused.target), game: game)
        let m = motion
        part(WaveText(L(.refusedTitle), height: 0.15, colour: C.pink, bob: 0.4, id: "refused_title_header", motion: m),
             at: at(0, top - 0.5))
        let card = part(Panel(size: [1.66, 1.2, 0.14], colour: C.paper, entrance: .tumble, motion: m), at: at(0, top - 1.45))
        child(Paragraph(L(.refusedBody), height: 0.07, colour: C.ink, width: 1.44, id: "refused_body", motion: m),
              at: at(0, 0.08), on: card.content).show(after: 0)
        // The reason, small, for whoever is asked to look at the phone.
        child(Label3D(why, height: 0.035, colour: C.disabledInk, maxWidth: 1.44, motion: m), at: at(0, -0.45),
              on: card.content).show(after: 0)
        let go = part(BlockButton(L(.refusedStartOver), id: "refused_startover_button", style: .danger, size: [1.3, 0.38],
                                  textHeight: 0.13, motion: m) { [weak game] in game?.startOver() }, at: at(0, bottom + 0.4))
        go.bobs = true
    }
}
